### PR TITLE
Fetch vote transactions from Arweave using GQL

### PR DESCRIPTION
**Pull request by KPrasch**

### CODE REVIEW

File: electora_py/api.py [hunk 1]
```diff
@@ -1,11 +1,33 @@
...
-from nucypher_core import ferveo
+
+# Networks
 
 _GOERLI_URI = "https://goerli.infura.io/v3/663d60ae0f504f168b362c2bda60f81c"
 _TEACHER_URI = "https://lynx.nucypher.network:9151"
 
+# GraphQL
+
+_ELECTORA_ARWEAVE_TAG = "electora/ballot/uuid"
```
Comment (KPrasch) [inline electora_py/api.py:17]: This tag format is a suggestion for some more semantic density, but it can be whatever!

File: electora_py/api.py [hunk 2]
```diff
@@ -17,7 +39,25 @@
...
+    result = client.execute(
+        gql(_GET_ELECTION_VOTES_QUERY_TEMPLATE),
+        variable_values=variables_map
+    )
+    return result
+
+
+def fetch_votes(election_id: str, endpoint: Optional[str] = _ARWEAVE_GQL_ENDPOINT):
+    transactions = _fetch_vote_transactions(election_id=election_id, endpoint=endpoint)
+    print(transactions)
```
Comment (meatballs) [inline electora_py/api.py:60]: Let's return these from here and add a cli command to do the print

File: electora_py/api.py (hunk 1 above)
Comment (theref) [reply electora_py/api.py:17]: in order to maintain future compatibility, i think I'm going to go with multiple tags:
```
tags = [{ "name": "ballot_uuid", "value": ballot.uuid },
        { "name": "application", "value": "electora"},
        # vote type,
        # version]
```
That way adding new metadata doesn't invalidate old stuff

File: electora_py/api.py [hunk 3]
```diff
@@ -1,11 +1,33 @@
...
+
+# Networks
 
 _GOERLI_URI = "https://goerli.infura.io/v3/663d60ae0f504f168b362c2bda60f81c"
 _TEACHER_URI = "https://lynx.nucypher.network:9151"
 
+# GraphQL
+
+_ELECTORA_ARWEAVE_TAG = "electora/ballot/uuid"
+_ARWEAVE_GQL_ENDPOINT = "https://arweave.net/graphql"  # TODO: Get this from somewhere else (config, env, round-robin, etc.)
```
Comment (meatballs) [inline electora_py/api.py:18]: Can we please not use comments for this? If something needs doing, create an issue for it.

File: electora_py/api.py [hunk 4]
```diff
@@ -1,11 +1,33 @@
...
+from gql import gql, Client
+from gql.transport.aiohttp import AIOHTTPTransport
+from nucypher_core import ferveo
+
+from nucypher.characters.lawful import Ursula, Bob
 from nucypher.cli.utils import connect_to_blockchain
 from nucypher.utilities.emitters import StdoutEmitter
-from nucypher_core import ferveo
+
+# Networks
```
Comment (meatballs) [inline electora_py/api.py:10]: Can we please avoid using comments unless they're explaining something that's not obvious?

File: electora_py/api.py [hunk 5]
```diff
@@ -1,11 +1,33 @@
...
 
 _GOERLI_URI = "https://goerli.infura.io/v3/663d60ae0f504f168b362c2bda60f81c"
 _TEACHER_URI = "https://lynx.nucypher.network:9151"
 
+# GraphQL
```
Comment (meatballs) [inline electora_py/api.py:15]: Again, let's not.

File: electora_py/api.py (hunk 2 above)
Comment (KPrasch) [reply electora_py/api.py:60]: The `print` and `pass` here are meant to indicate the this is still stub functionality (the transaction results need further processing before we actually have _votes_ in scope).  Do you have a preferred way to indicate a stub?

File: electora_py/api.py (hunk 4 above)
Comment (KPrasch) [reply electora_py/api.py:10]: If you'd prefer to not have a comment here, I'll remove it.  I think of this as a logical break appropriate for the current development stage.

**Review by theref — APPROVED** [review]

File: electora_py/api.py (hunk 3 above)
Comment (KPrasch) [reply electora_py/api.py:18]: I suppose it's a principled debate, but IMO this indictes the exact entry point that a change needs to occur (in fact I'd like to update this comment to include the issue number).  Anyways, I'll remove it.  Opened #2.

File: electora_py/api.py [hunk 6]
```diff
@@ -17,7 +39,25 @@
...
+        gql(_GET_ELECTION_VOTES_QUERY_TEMPLATE),
+        variable_values=variables_map
+    )
+    return result
+
+
+def fetch_votes(election_id: str, endpoint: Optional[str] = _ARWEAVE_GQL_ENDPOINT):
+    transactions = _fetch_vote_transactions(election_id=election_id, endpoint=endpoint)
+    print(transactions)
     pass
```
[the remark this answers is not in the document]
Comment (KPrasch) [reply electora_py/api.py]: 👌🏻

File: electora_py/api.py (hunk 4 above)
Comment (meatballs) [reply electora_py/api.py:10]: I loathe comments. They get chucked in, don't get maintained  and just add bloat to a codebase. Unless they're explaining something that's tricky to get your head round, they should be avoided. (Even then, refactor out to a function and give it a decent docstring is often a better option).

File: electora_py/api.py (hunk 2 above)
Comment (meatballs) [reply electora_py/api.py:60]: I'd raise a `NotImplementedError`

Comment (KPrasch) [reply electora_py/api.py]: ✔️

File: electora_py/api.py (hunk 1 above)
Comment (KPrasch) [reply electora_py/api.py]: Opened https://github.com/electora-vote/electora-py/issues/4

**Pull request merged**